designs/PackageVulnerability/FlagVulnerablePackages.md

```diff
@@ -0,0 +1,147 @@
+* This feature is an enhancement to the NuGet v3 protocol. We won't surface this in earlier versions of the NuGet protocol.
+* `NuGet.Server` is out-of-scope for this feature at this point.
+
+## Scope
+
+There are multiple scenarios for flagging and reporting vulnerabilities. This feature will be implemented in several stages which will enable a subset of those as discussed below.
+
+### Stage 1 - MVP
+
+For the [MVP](https://en.wikipedia.org/wiki/Minimum_viable_product) of this feature, we'll focus on:
+
+* Enabling ***consumers** to be informed about known vulnerabilities before installation* of a package.
+* Enabling ***authors** to be informed about known vulnerabilities in their packages*.
+* Leveraging **existing, curated, and publicly available known vulnerabilities** affecting NuGet packages.
+
+This should help prevent *new* installs of packages that are known to be vulnerable, and trigger authors into providing a fix through a new package version on nuget.org.
+
+NuGet will not support self-reporting of vulnerabilities at this time and instead rely on an existing data provider of curated vulnerability information. For the MVP of this feature, all vulnerability data will be sourced from external systems. The initial focus is on leveraging [GitHub's Security Workflow](https://github.com/features/security) and integration with [GitHub's GraphQL API](https://developer.github.com/v4/). This allows us to build an initial end-to-end experience without having to deal with the complexities of curating vulnerability reports.
+
+#### ✔️ In scope (Stage 1)
+
+* For CLI:
+  * A CLI command to **list vulnerabilities** in a project/solution project graph
+
+* For Visual Studio:
+  * **Display vulnerability information** in the package graph on `Installed` tab of Package Manager Dialog (PMUI)
+
+* For nuget.org:
+  * **Automatically ingest and refresh NuGet package vulnerability data from GitHub**
+  * **Surface package vulnerability information on-demand**:
+    * as part of package metadata in `v3` registration blobs (`JSON`)
+    * by showing vulnerability information to package owners (only!) on package details page and version history
+  * Basic (e-mail) notification to notify package publishers for any new vulnerability detected in one or more of their packages
+
+#### ❌ Out of scope (Stage 1)
+
+Some enhancements to this feature are out of scope for the initial implementation of this feature because there are several unknowns still to be clarified. See also the *Open Questions* section below.
+
+* Enable `restore` to Inform or Warn (see Stage 2)
+* Expose vulnerability metadata in nuget.org *Search* service (not required for MVP)
+* Display vulnerability information for *transitive packages* in PMUI (additional UX complexity)
+* Expose vulnerability information *publicly* on nuget.org's package details page and version history (disclosure policies TBD)
+
+### Stage 2
```

Comment:
It would be good if we could move Stage 2 to another design spec that we can review when we are ready to take on that work. As part of this spec, we should only mention potential enhancements in Stage 2 that have an impact on the initial design.